src/TestFramework/PhpUnit/Coverage/IndexXmlCoverageReader.php

```diff
@@ -33,32 +33,33 @@
 
 declare(strict_types=1);
 
-namespace Infection\TestFramework\Coverage\XmlReport;
+namespace Infection\TestFramework\PhpUnit\Coverage;
 
-use Infection\TestFramework\Coverage\SourceFileDataProvider;
-use Infection\TestFramework\PhpUnit\Coverage\IndexXmlCoverageParser;
+use const DIRECTORY_SEPARATOR;
+use function Safe\file_get_contents;
 
 /**
  * @internal
+ * @final
  */
-final class FileCodeCoverageProviderFactory
+class IndexXmlCoverageReader
 {
+    private const COVERAGE_INDEX_FILE_NAME = 'index.xml';
+
     private $coverageDir;
-    private $coverageXmlParser;
 
-    public function __construct(
-        string $coverageDir,
-        IndexXmlCoverageParser $coverageXmlParser
-    ) {
+    public function __construct(string $coverageDir)
+    {
         $this->coverageDir = $coverageDir;
-        $this->coverageXmlParser = $coverageXmlParser;
     }
 
-    public function create(): SourceFileDataProvider
+    public function getIndexXmlPath(): string
+    {
+        return $this->coverageDir . DIRECTORY_SEPARATOR . self::COVERAGE_INDEX_FILE_NAME;
```

Comment:
Can't we return a canonalized path here and memoize the call?

Reply:
I'm sure we can, but for a method that is called only two times per run, is it worth it?

Reply:
OK, maybe four times if include CoverageChecker. I'm not so sure about memorization here.